### DISCUSSION TITLE
local type alias

### DISCUSSION THREAD

**Discussion by stephentoub** [dotnet/csharplang#445] [General]
_From @juliusfriedman on July 19, 2016 14:40_

Currently class declarations do not allow nested type aliasing e.g.

`using MyTuple = System.Tuple<int, int>;` even within a namespace declaration...

This declaration must normally be at the top of a file and precede all other using statements such as `extern`.

I propose that this restriction be lessened such that to allow one to use such aliases within their definitions to avoid redundant typing.

The only way this can be achieved now is to separate each class into it's own file and definitions of the class which fulfill the rule of the compiler by using `partial` or by declaring their own type which represents the type layout desired [thus reducing the use of a perfectly good structure such as `Tuple` simply due to the anonymity associated with such structures.].

Thank you for your time and consideration.

_Copied from original issue: dotnet/corefx#10150_

**Comment by alrz** [comment]
Probably the same as https://github.com/dotnet/csharplang/issues/218
